app/scripts/migrations/054.js

```diff
@@ -0,0 +1,74 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+function isValidDecimals(decimals) {
+  return (
+    typeof decimals === 'number' ||
+    (typeof decimals === 'string' && decimals.match(/^(0x)?\d+$/u))
+  );
+}
+
+/**
+ * Migrates preference tokens with 0 decimals typed as 'string' to 'number'
```

Comment:
Nit: Just noticed this comment was out of date too!

Reply:
fixed!